### ISSUE TITLE
[chip-test] chip_sw_data_integrity

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#14128]
### Test point name

[chip_sw_data_integrity](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/data/chip_testplan.hjson#L453)

### Host side component

SystemVerilog

### OpenTitanTool infrastructure implemented

No

### Contact person

tjaychen

### Checklist

Please fill out this checklist as items are completed. Link to PRs and issues as appropriate.
- [ ] Check if existing test covers most or all of this testpoint (if so, either extend said test to cover all points, or skip the next 3 checkboxes)
- [ ] Device-side (C) component developed
- [ ] Bazel build rules developed
- [ ] Host-side component developed
- [ ] HJSON test plan updated with test name (so it shows up in the dashboard)
- [ ] Test added to dvsim nightly regression (and passing at time of checking)

**Comment by weicaiyang** [comment]
@matutem I think your escalation test can cover this, so I assign this to you.

**Comment by matutem** [comment]
Yes, it covers most of this, except we want to add an extra test that introduces sram integrity errors to verify that the CPU will trigger the integrity violation.
